tests/ui/lint/invalid_value.stderr

```diff
@@ -427,18 +417,14 @@ error: the type `OneFruitNonZero` does not permit being left uninitialized
   --> $DIR/invalid_value.rs:107:37
    |
 LL |         let _val: OneFruitNonZero = mem::uninitialized();
-   |                                     ^^^^^^^^^^^^^^^^^^^^
-   |                                     |
-   |                                     this code causes undefined behavior when executed
-   |                                     help: use `MaybeUninit<T>` instead, and only call `assume_init` after initialization is done
+   |                                     ^^^^^^^^^^^^^^^^^^^^ this code causes undefined behavior when executed
    |
    = note: `OneFruitNonZero` must be non-null
 note: because `std::num::NonZero<u32>` must be non-null (in this field of the only potentially inhabited enum variant)
   --> $DIR/invalid_value.rs:38:12
    |
 LL |     Banana(NonZero<u32>),
    |            ^^^^^^^^^^^^
-   = note: integers must be initialized
```

Comment:
@oli-obk, maybe this can also be fixed with `normalize_erasing_regions`?

Reply:
Doesn't seem as straight-forward, since it then also exposes the implementation detail:

```
note: because `core::num::nonzero::private::NonZeroU32Inner` must be non-null
note: integers must be initialized
```

Reply:
That seems ok for a low level diagnostic like that

Reply:
Okay, I added this change and rebased the PR again to not have commits in the future.